File: BaekjoonWithSwift/BaekjoonWithSwift/정렬1/좌표정렬하기2.swift
//
//  좌표정렬하기2.swift
//  BaekjoonWithSwift
//
//  Created by kyungmin on 2024/01/09.
//

import Foundation

func 좌표정렬하기2() {
    let N = Int(readLine()!)!
    var numbers = [(Int, Int)]()
    for _ in 0..<N {
        let number = readLine()!.split(separator: " ").compactMap { Int($0) }
        numbers.append((number[0], number[1]))
    }
    
    numbers.sort { ($0.1, $0.0) < ($1.1, $1.0) }
    
    for (x, y) in numbers {
        print("\(x) \(y)")
    }
}

File: BaekJoon/정렬 1/좌표 정렬하기2.py
"""
author: kyungmin
date: 24.1.9
title: 좌표 정렬하기2
time: 30분
"""
import sys
input = sys.stdin.readline

N = int(input().rstrip())
numbers = []

for _ in range(N):
    x, y = map(int, input().rstrip().split())
    numbers.append((x, y))

numbers.sort(key = lambda x: (x[1], x[0]))
for (x, y) in numbers:
    print(f"{x} {y}")